File: app/src/main/java/com/wolfpeng/androidframework/base/BaseFragment.java
package com.wolfpeng.androidframework.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * author：WolfWang
 * date：2017/11/7 16:10
 * e-mail：1678173987@qq.com
 * description：
 */

public  abstract  class BaseFragment extends Fragment {



    public abstract int getLayoutId();
    Unbinder unbinder;
    public abstract void initData(View view, @Nullable Bundle savedInstanceState);

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view =inflater.inflate(getLayoutId(),container,false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }


    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initData(view,savedInstanceState);
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/wolfpeng/androidframework/base/BaseFragment.kt
package com.wolfpeng.androidframework.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup


/**
 * author：WolfWang
 * date：2017/9/25 13:17
 * e-mail：1678173987@qq.com
 * description：
 */
class BaseFragment:Fragment(){


    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return super.onCreateView(inflater, container, savedInstanceState)
    }





}